src/cargo/core/compiler/mod.rs

```diff
@@ -683,6 +700,9 @@ fn add_error_format_and_color(
         _ => {}
     }
     cmd.arg(json);
+    if cx.bcx.config.cli_unstable().jobserver_per_rustc {
+        cmd.arg("-Zjobserver-token-requests");
```

Comment:
Could this perhaps be joint with the `jobserver_per_rustc` read above? (consolidating the `if` statements)